model/infinispan/src/main/java/org/keycloak/spi/infinispan/impl/Util.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2025 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.spi.infinispan.impl;
+
+import org.keycloak.config.Option;
+import org.keycloak.provider.ProviderConfigurationBuilder;
+
+/**
+ * Utility method for this package and subpackages
+ */
+public final class Util {
+
+    private Util() {
+    }
+
+    /**
+     * Copies the {@link Option} information into the {@link ProviderConfigurationBuilder}.
+     *
+     * @param builder  The property to set/configure.
+     * @param name     The desired property name.
+     * @param label    The label of the property's argument.
+     * @param type     The type of the property's value.
+     * @param option   The source {@link Option} to gather the information.
+     * @param isSecret {@code true} if the property is a secret.
+     */
+    public static void copyFromOption(ProviderConfigurationBuilder builder, String name, String label, String type, Option<?> option, boolean isSecret) {
```

Comment:
This is currently only called from `DefaultCacheRemoteConfigProviderFactory` do you have plans to use it elsewhere? If not, maybe we should just leave in `DefaultCacheRemoteConfigProviderFactory` and move out if/when required.

Reply:
It's going to be used by the embedded config provider. Work in progress in this branch:
https://github.com/pruivo/keycloak/blob/f783965e234be052919201c16e2cea0e7d69ee6a/model/infinispan/src/main/java/org/keycloak/spi/infinispan/impl/embedded/DefaultCacheEmbeddedConfigProviderFactory.java#L217-L226